include/rime_api.h

```diff
@@ -483,6 +483,9 @@ typedef struct rime_api_t {
   //! caret posistion in terms of raw input
   size_t (*get_caret_pos)(RimeSessionId session_id);
 
+  //! set caret posistion in terms of raw input
+  void (*set_caret_pos)(RimeSessionId session_id, size_t caret_pos);
```

Comment:
If you insert a function pointer to the API struct, you break compatibility of binaries compiled against the previous librime version. By design, new API functions can only be added at the end of struct.